cluster/service.go

```diff
@@ -211,9 +235,10 @@ func (s *Service) handleConn(conn net.Conn) {
 			stats.Add(numExecuteRequest, 1)
 
 			resp := &CommandExecuteResponse{}
-
 			er := c.GetExecuteRequest()
-			if er == nil {
+			if !s.checkCommandPerm(c, auth.PermExecute) {
+				resp.Error = "Unauthorized"
+			} else if er == nil {
```

Comment:
Style thing, I realise, but I think the `er == nil` check should go first.

Reply:
I'm definitely fine with changing this and will (not dogmatic about it in the least) but I am curious about why you prefer that choice (even stylistically). My thinking here is basically return auth failures as soon as you can, but admittedly I didn't actually do completely that because I wanted to put this check in the existing if/else block rather than do the check before we even extract the ExecuteRequest.

Reply:
Two reasons.

- I like to check fundamental assumptions first. If the `er` is nil, something has gone fundamentally wrong with the call.
- Secondly, this code reminds me of Go error checking, and the `if err == nil` pattern. So to my eyes, it looks more Go-like to check a nil case first.

Again, just style.